cmd/jsonnet.cpp

```diff
@@ -455,44 +456,53 @@ static bool process_args(int argc,
         return false;
     }
 
-    std::string filename = remaining_args[0];
-    if (remaining_args.size() > 1) {
-        std::cerr << "ERROR: Already specified " << want
-                  << " as \"" << filename << "\"\n"
-                  << std::endl;
-        usage(std::cerr);
+    bool multiple_files_allowed = config->cmd == FMT &&
+                                  (config->fmtTest || config->fmtInPlace);
+    if (!multiple_files_allowed) {
+        std::string filename = remaining_args[0];
+        if (remaining_args.size() > 1) {
+            std::cerr << "ERROR: Already specified " << want
+                      << " as \"" << filename << "\"\n"
+                      << std::endl;
+            usage(std::cerr);
+            return false;
+        }
+    }
+    config->inputFiles = remaining_args;
+    return true;
+}
+
+static bool read_input_contents(std::string filename, std::string *input) {
+    std::ifstream f;
+    f.open(filename);
+    if (!f.good()) {
+        std::string msg = "Opening input file: " + filename;
+        perror(msg.c_str());
+        return false;
+    }
+    input->assign(std::istreambuf_iterator<char>(f),
+                  std::istreambuf_iterator<char>());
+    if (!f.good()) {
+        std::string msg = "Reading input file: " + filename;
+        perror(msg.c_str());
         return false;
     }
-    config->inputFile = filename;
     return true;
 }
 
 /** Reads Jsonnet code from the input file or stdin into the input buffer. */
-static bool read_input(JsonnetConfig* config, std::string* input) {
+static bool read_input(JsonnetConfig* config, int index, std::string* input) {
```

Comment:
It might be neater to take a pointer to the input file you're going to read then, unless you think there is an imminent use case for taking `index` that's not shown yet